connect/runtime/src/main/java/org/apache/kafka/connect/runtime/rest/admin/LoggingResource.java

```diff
@@ -0,0 +1,173 @@
+
+    /**
+     * List the current loggers that have their levels explicitly set and their log levels.
+     *
+     * @return a list of current loggers and their levels.
+     */
+    @GET
+    @Path("/")
+    public Response listLoggers() {
+        Enumeration en = LogManager.getCurrentLoggers();
+        Map<String, Map<String, String>> loggers = new TreeMap<>();
+        while (en.hasMoreElements()) {
+            Logger current = (Logger) en.nextElement();
+            // exclude loggers that do not have a specific level set
+            if (current.getLevel() != null) {
+                loggers.put(current.getName(), levelToMap(current.getLevel()));
+            }
+        }
+
+        Logger root = LogManager.getRootLogger();
+        if (root.getLevel() != null) {
+            loggers.put(ROOT_LOGGER_NAME, levelToMap(root.getLevel()));
+        }
+
+        return Response.ok(loggers).build();
+    }
+
+    /**
+     * Get the log level of a named logger.
+     *
+     * @param namedLogger name of a logger
+     * @return level of the logger, effective level if the level was not explicitly set.
+     */
+    @GET
+    @Path("/{logger}")
+    public Response getLogger(final @PathParam("logger") String namedLogger) {
+        Objects.requireNonNull(namedLogger, "require non-null name");
+
+        Enumeration en = LogManager.getCurrentLoggers();
+        Logger logger = null;
+        while (en.hasMoreElements()) {
+            Logger l = (Logger) en.nextElement();
+            if (namedLogger.equals(l.getName())) {
+                logger = l;
+                break;
+            }
+        }
+        if (logger == null) {
+            throw new NotFoundException("logger " + namedLogger + " not found.");
+        } else {
+            Map<String, String> levelMap = new HashMap<>();
+
+            String level;
+            if (logger.getLevel() == null) {
+                level = String.valueOf(logger.getEffectiveLevel());
+            } else {
+                level = String.valueOf(logger.getLevel());
+            }
+
+            levelMap.put("level", level);
+            return Response.ok(levelMap).build();
+        }
+    }
+
+
+    /**
+     * Adjust level of a named logger. if name corresponds to an ancestor, then the log level is applied to all child loggers.
+     *
+     * @param namedLogger name of the logger
+     * @param levelMap a map that is expected to contain one key 'level', and a value that is one of the log4j levels:
+     *                 DEBUG, ERROR, FATAL, INFO, TRACE, WARN
+     * @return names of loggers whose levels were modified
+     */
+    @PUT
+    @Path("/{logger}")
+    @SuppressWarnings("unchecked")
+    public Response setLevel(final @PathParam("logger") String namedLogger,
+                             final Map<String, String> levelMap) {
+        String desiredLevelStr = levelMap.get("level");
+        Objects.requireNonNull(desiredLevelStr, "desired level was not specified in PUT request.");
```

Comment:
Do we want to fail with a NPE if the payload does not contain `level`? Would it be better to use `BadRequestException`? Also, the error message should be a bit more clear about exactly what was missing.